posthog/cache_utils.py

```diff
@@ -16,13 +17,28 @@ def memoized_fn(*args, use_cache=not TEST, **kwargs):
                 return fn(*args, **kwargs)
 
             current_time = now()
-
             key = (args, frozenset(sorted(kwargs.items())))
-            if key not in memoized_fn._cache or current_time - memoized_fn._cache[key][0] > cache_time:
-                memoized_fn._cache[key] = (current_time, fn(*args, **kwargs))
+
+            def refresh():
+                value = fn(*args, **kwargs)
+                memoized_fn._cache[key] = (now(), value)
+                memoized_fn._refreshing[key] = None
```

Comment:
Should we capture any exceptions in `fn` to ensure we set `memoized_fn._refreshing[key] = None`?

Is it possible this `redis_client.zrangebyscore` call fails? Perhaps due to slow/intermittent connection issues. This would cause us to leave `memoized_fn._refreshing[key]` set to `current_time` forever.

Reply:
Thats a good call 👍